include/i2s.h

```diff
@@ -0,0 +1,418 @@
+ *     ---.---.---.---.---.---.---.---.---.---.---.---.---.---.---.---.---.---.-
+ * SD     |MSB|   |...|   |LSB| x |...| x |MSB|   |...|   |LSB| x |...| x |   |
+ *     ---'---'---'---'---'---'---'---'---'---'---'---'---'---'---'---'---'---'-
+ *        | Left channel                  | Right channel                 |
+ * @endverbatim
+ */
+#define I2S_FMT_DATA_FORMAT_LEFT_JUSTIFIED  (3 << I2S_FMT_DATA_FORMAT_SHIFT)
+
+/**
+ * @brief Right Justified Data Format.
+ *
+ * Serial data is transmitted in two’s complement with the MSB first. Both
+ * Word Select (WS) and Serial Data (SD) signals are sampled on the rising edge
+ * of the clock signal (SCK). The bits within the data word are right justified
+ * such that the LSB is always sent in the clock period preceding the WS
+ * transition. Left channel data are sent first indicated by WS = 1, followed
+ * by right channel data indicated by WS = 0.
+ *
+ * @verbatim
+ *      .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-. .-.
+ * SCK -' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-' '-
+ *        .-------------------------------.                               .-----
+ * WS  ---'                               '-------------------------------'
+ *     ---.---.---.---.---.---.---.---.---.---.---.---.---.---.---.---.---.---.-
+ * SD     | x |...| x |MSB|   |...|   |LSB| x |...| x |MSB|   |...|   |LSB|   |
+ *     ---'---'---'---'---'---'---'---'---'---'---'---'---'---'---'---'---'---'-
+ *        | Left channel                  | Right channel                 |
+ * @endverbatim
+ */
+#define I2S_FMT_DATA_FORMAT_RIGHT_JUSTIFIED (4 << I2S_FMT_DATA_FORMAT_SHIFT)
+
+/** Send MSB first */
+#define I2S_FMT_DATA_ORDER_MSB              (0 << 3)
+/** Send LSB first */
+#define I2S_FMT_DATA_ORDER_LSB              (1 << 3)
+/** Invert bit ordering, send LSB first */
+#define I2S_FMT_DATA_ORDER_INV              I2S_FMT_DATA_ORDER_LSB
+/** Invert bit clock */
+#define I2S_FMT_BIT_CLK_INV                 (1 << 4)
+/** Invert frame clock */
+#define I2S_FMT_FRAME_CLK_INV               (1 << 5)
+
+
+typedef u8_t i2s_opt_t;
+
+/** Run bit clock continuously */
+#define I2S_OPT_BIT_CLK_CONT                (0 << 0)
+/** Run bit clock when sending data only */
+#define I2S_OPT_BIT_CLK_GATED               (1 << 0)
+/** I2S driver is bit clock master */
+#define I2S_OPT_BIT_CLK_MASTER              (0 << 1)
+/** I2S driver is bit clock slave */
+#define I2S_OPT_BIT_CLK_SLAVE               (1 << 1)
+/** I2S driver is frame clock master */
+#define I2S_OPT_FRAME_CLK_MASTER            (0 << 2)
+/** I2S driver is frame clock slave */
+#define I2S_OPT_FRAME_CLK_SLAVE             (1 << 2)
+/** @brief Loop back mode.
+ *
+ * In loop back mode RX input will be connected internally to TX output.
+ * This is used primarily for testing.
+ */
+#define I2S_OPT_LOOPBACK                    (1 << 7)
+
+
+typedef u32_t i2s_dir_t;
```

Comment:
I didn't use enum as I2S_DIR_RX, I2S_DIR_TX may be combined with a bitwise operator, e.g. (I2S_DIR_RX | I2S_DIR_TX) to perform operation on both directions simultaneously. Enum would have to be defined as:
```
enum i2s_dir {
    I2S_DIR_RX,
    I2S_DIR_TX,
    I2S_DIR_RX_TX,
};
```
That's also an option. Should I change it?

As for the typedef let's keep it in line with i2s_opt_t, i2s_fmt_t. If my reasons for using specific types (as stated above) are not good enough I'll remove it from the API.